script/blockchains-config/add-network-rpc.ts

```diff
@@ -0,0 +1,100 @@
+import 'dotenv/config'
+import { defineCommand, runMain } from 'citty'
+import { MongoClient } from 'mongodb'
+import consola from 'consola'
+
+const main = defineCommand({
+  meta: {
+    name: 'add-network-rpc',
+    description: 'Add a new RPC endpoint to the blockchain configuration',
+  },
+  args: {
+    network: {
+      type: 'string',
+      description: 'Network name to which the RPC endpoint should be added',
+      required: true,
+    },
+    rpcUrl: {
+      type: 'string',
+      description: 'RPC URL of the endpoint to add',
+      required: true,
+    },
+    environment: {
+      type: 'string',
+      description: 'Environment of the RPC endpoint (default is production)',
+      required: false,
+      default: 'production',
+    },
+  },
+  async run({ args }) {
+    const { network, rpcUrl, environment } = args
+    // Use the provided network as the chainName
+    const chainName = network
+
+    // Connect to MongoDB using the MONGODB_URI from environment variables
+    const MONGODB_URI = process.env.MONGODB_URI as string
+    if (!MONGODB_URI) {
+      consola.error('MONGODB_URI is not defined in the environment')
+      process.exit(1)
+    }
+
+    const client = new MongoClient(MONGODB_URI)
+    await client.connect()
+    const db = client.db('blockchain_configs')
+    const collection = db.collection('rpc_endpoints')
+
+    // Check if there's an existing document for the given chainName
+    const existingDoc = await collection.findOne({ chainName })
+
+    // Calculate the new highest priority: if there are existing endpoints, assign (max priority + 1), else 1
+    let newPriority = 1
+    if (
+      existingDoc &&
+      Array.isArray(existingDoc.rpcs) &&
+      existingDoc.rpcs.length > 0
+    ) {
+      newPriority =
+        Math.max(
+          ...existingDoc.rpcs.map(
+            (rpc: { priority: number }) => rpc.priority || 0
+          )
+        ) + 1
+    }
+
+    // Check if the RPC endpoint already exists for the given chain
```

Comment:
Would it make sense to update the priority if an RPC endpoint is already known?
That way we could use this function to change the priorities (e.g. temporarily use another one, then switch back to the original).
Right now we would have to manually manipulate the entries in the database and if we would update the priority here we could do it using this command.

Reply:
feel free to resolve that conversation

Reply:
deleted display-rpcs as it will be reflected in .env when fetching rpcs in exactly what you wrote in the comment below